### PR TITLE
Adding VPP Unit tests

### PR DESCRIPTION
**Pull request by Eoghan1232**
Added unit tests for vpp
Added new Dockerfile for unit test container to run Fixed bug found in code
fixed some logging to return as a useful error when failing

### CODE REVIEW

File: cnivpp/cnivpp_test.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+			sharedDir, dirErr := os.MkdirTemp("/tmp", "test-cnivpp-")
+			require.NoError(t, dirErr, "Can't create temporary directory")
+			defer os.RemoveAll(sharedDir)
+
+			pod := GetTestPod(sharedDir)
+			kubeClient := fake.NewSimpleClientset(pod)
+
+			// SetExecCommand(&FakeExecCommand{Err: tc.fakeErr})
+			err := cniVpp.AddOnHost(tc.netConf, args, kubeClient, sharedDir, result)
+			// SetDefaultExecCommand()
```
Comment (johnoloughlin) [inline cnivpp/cnivpp_test.go:234]: can this be removed?

File: cnivpp/cnivpp_test.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+			args := GetTestArgs()
+
+			sharedDir, dirErr := os.MkdirTemp("/tmp", "test-cnivpp-")
+			require.NoError(t, dirErr, "Can't create temporary directory")
+			defer os.RemoveAll(sharedDir)
+
+			pod := GetTestPod(sharedDir)
+			kubeClient := fake.NewSimpleClientset(pod)
+
+			// SetExecCommand(&FakeExecCommand{Err: tc.fakeErr})
```
Comment (johnoloughlin) [inline cnivpp/cnivpp_test.go]: can this be removed?

File: cnivpp/cnivpp_test.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+
+			err := cniVpp.DelFromHost(tc.netConf, args, sharedDir)
+			// time.Sleep(time.Second * 999999)
+			if tc.expErr == nil {
+				assert.Equal(t, tc.expErr, err, "Unexpected result")
+			} else {
+				require.Error(t, err, "Unexpected result")
+				assert.Contains(t, err.Error(), tc.expErr.Error(), "Unexpected result")
+			}
+			// time.Sleep(time.Second * 999999)
```
Comment (johnoloughlin) [inline cnivpp/cnivpp_test.go:317]: can this be removed?

Comment (Eoghan1232) [reply cnivpp/cnivpp_test.go:317]: done.

**Review by johnoloughlin — APPROVED** [review]

**Pull request merged**